FileMapper/API_util.js

```diff
@@ -0,0 +1,435 @@
+
+  var path = mappingObj.drivepath;
+
+  // Drive path is connected by forward slash (hence isUnix is set true)
+  var isUnix = true;
+  var isFile = !mappingObj.isfolder;
+  var index = PathUtil.getFirstSlash(path, isUnix);
+  // Removing the "My Drive" part from the path since it is the root folder and
+  // will be used as the current directory
+  path = path.slice(index + 1);
+
+  var driveId = createInDrive(DriveApp.getRootFolder(), path, isUnix, isFile);
+
+  mappingObj.id = driveId;
+  CONFIG.setMappingInConfigData(localPath, mappingObj);
+  CONFIG.flushConfigDataToFile();
+  return driveId;
+}
+
+/**
+ * Create the destination file/folder in the drive by traversing the given path
+ *
+ * @param {FolderObject} currentDirectory The Folder object for the current
+ *     directory
+ * @param {String} relativePath The path to be traversed from the current
+ *     directory to reach the destination
+ * @param {boolean} isUnix To signify whether its a windows path or unix path
+ * @param {boolean} isFile To signify whether its a file or folder
+ * @return {String} driveId The drive id of the local file/folder if found or
+ *     null if file is not found
+ */
+function createInDrive(currentDirectory, relativePath, isUnix, isFile) {
+  var index = PathUtil.getFirstSlash(relativePath, isUnix);
+
+  // If index is -1 means we are at the destination file/folder
+  if (index === -1) {
+    return ApiUtil.createDestination(relativePath, currentDirectory, isFile);
+  } else {
+    // The folder which is to be searched in current directory
+    var folderName = relativePath.slice(0, index);
+    // New path is now without the folder which will be searched in this call
+    relativePath = relativePath.slice(index + 1);
+
+    var folders = currentDirectory.getFoldersByName(folderName);
+    var folder = (folders.hasNext())
+                     ? folders.next()
+                     : ApiUtil.searchFolderByName(folderName, currentDirectory);
+    if (folder === null) {
+      folder = currentDirectory.createFolder(folderName);
+    }
+
+    return ApiUtil.createInDrive(folder, relativePath, isUnix, isFile);
+  }
+}
+
+/**
+ * Creates the destination file/folder having same mimetype as the local file
+ *
+ * @param {String} destinationName The name of file/folder which is to be
+ *     created
+ * @param {FolderObject} folder The Folder in which the file is to be created
+ * @param {boolean} isFile To signify whether its a file or folder
+ * @return {String} driveId The drive id of the local file/folder created
+ */
+function createDestination(destinationName, folder, isFile) {
+  var destination;
+  var driveId;
+  if (isFile) {
+    var extension = PathUtil.getExtension(destinationName);
+    // Making the local file without extension to be text file
+    if (extension === "") {
+      extension = "txt";
+    }
+
+    var mimetype = ConfigUtil.getMimeTypeFromExtension(extension);
+    switch (mimetype) {
+      case MimeType.GOOGLE_SHEETS: {
```

Comment:
No need for braces for each case, it's ok but not generally used

Reply:
Okay. Fixed.